src/main/java/fi/otavanopisto/kuntaapi/server/integrations/casem/CaseMCacheUpdater.java

```diff
@@ -0,0 +1,847 @@
+        model.setDecision(text);
+      break;
+      case EXTENDED_NAME:
+        model.setName(text);
+      break;
+      case EXTENDED_DECISION_NOTES:
+        model.setDecisionNotes(text);
+      break;
+      case EXTENDED_DECISIONPROPOSAL:
+        model.setDecisionProposal(text);
+      break;
+      case EXTENDED_CORRECTIONINSTRUCTIONS:
+        model.setCorrectioninstructions(text);
+      break;
+      case EXTENDED_IS_ADDITIONAL_TOPIC:
+        model.setAdditionalTopic("True".equals(text));
+      break;
+      case EXTENDED_HISTORY_TOPICS:
+        model.setHistoryTopics(parseHistoryTopics(text));
+      break;
+      default:
+        logger.warning(String.format("Unrecognized extended meeting item property %s", name));
+      break;
+    }
+  }
+  
+  private String parseAttachments(String downloadUrl, String text) {
+    String result = text;
+    Pattern pattern = Pattern.compile("(download.aspx\\?ID=)([0-9]*)\\&GUID=\\{([0-9a-zA-Z-]*)\\}");
+    Matcher matcher = pattern.matcher(result);
+    while (matcher.find()) {
+      Long id = NumberUtils.createLong(matcher.group(2));
+      String guid = matcher.group(3);
+      String url = String.format("%s/{%s}/%d", downloadUrl, guid, id);
+      result = matcher.replaceFirst(url);
+      matcher = pattern.matcher(result);
+    }
+    
+    return result;
+  }
+
+  private List<ExtendedProperty> listExtendedProperties(OrganizationId organizationId, Content content) {
+    ApiResponse<ExtendedPropertyList> propertiesResult = caseMApi.getContentsApi(organizationId).listExtendedPropertiesByContent(content.getContentId(), null);
+    if (!propertiesResult.isOk()) {
+      logger.severe(String.format("Extended property listing failed on [%d] %s", propertiesResult.getStatus(), propertiesResult.getMessage()));
+      return Collections.emptyList();
+    }
+    
+    return propertiesResult.getResponse().getValue();
+  }
+  
+  private List<Participant> parsePresenters(String text) {
+    List<Participant> result = new ArrayList<>();
+    
+    List<String[]> lines = parseCaseMCSV(text);
+    for (String[] line : lines) {
+      if (line.length > 0) {
+        Participant participant = new Participant();
+        participant.setName(line[0]);
+        if (line.length > 1) {
+          participant.setTitle(line[1]);
+        }
+
+        if (line.length > 2) {
+          participant.setEmail(line[2]);
+        }
+        
+        result.add(participant);
+      }
+    }
+    
+    return result;
+  }
+
+  private Councilmen parseCouncilmen(String text) {
+    Councilmen result = new Councilmen();
+    
+    List<String[]> lines = parseCaseMCSV(text);
+    for (String[] line : lines) {
+      if (line.length > 5) {
+        logger.severe(String.format("Unexpected cell count %d", line.length));
+      } else {
+        parseCouncilman(result, line);
+      }
+    }
+    
+    return result;
+  }
+
+  private void parseCouncilman(Councilmen result, String[] line) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) The Cyclomatic Complexity of this method "parseCouncilman" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)